File: w1_p2/36.java
import java.util.Scanner;

class Main {
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		
		
		while (sc.hasNextLine()) {
			String input = sc.nextLine();
			
			if (input.charAt(0) == '0') {
				int hex = Integer.parseInt(input.substring(2), 16);
				System.out.println(hex);
			}
			else {
				int dec = Integer.parseInt(input);
				String hs = Integer.toHexString(dec);
				System.out.println("0x" + hs.toUpperCase());
			}
		}
	}
}

File: w1_p2/36.c
#include <stdio.h>
#define MAX_NUMBER 1024
int main(void) {
	char input[MAX_NUMBER];
	
	while(fgets(input, MAX_NUMBER, stdin) != NULL) {
		if(input[0] == '0') {
			int hex = 0;
			sscanf(input + 2, "%x", &hex);
			printf("%d\n", hex);
		}
		else {
			int dec = 0;
			sscanf(input, "%d", &dec);
			printf("0x%X\n", dec);
		}
	}
		
	return 0;
}
